### PR TITLE
implement globbing for WDL/WOM

### PR DESCRIPTION
**Pull request by danbills**
We've introduced a method on Wom's `Callable`:

`callWithInputs` that takes the parameters required to build the wom callable.

Under the hood, the TaskDefinition has the ability to glob files and should propagate that ability to be used in cromwell. 

Further, this globbing ability depends on the language.  So we now require the caller to TaskDefinition to provide the implementation of `globFiles`, which is defined as `Map[String, WdlValue] => Set[WdlGlobFile]`.

### CODE REVIEW

**Comment by danbills** [comment]
Adding do not merge as it makes sense to pull in @Horneth 's changes before this and re-basing.

File: wom/src/main/scala/wdl4s/wom/callable/TaskDefinition.scala [hunk 1]
```diff
@@ -18,8 +20,11 @@ case class TaskDefinition(name: String,
                           parameterMeta: Map[String, String],
                           outputs: Set[Callable.OutputDefinition],
                           inputs: List[_ <: Callable.InputDefinition],
+                         //TODO: Move this out?  Not being used.  Or add prefix separator field on Callable
```
Comment (Horneth) [inline wom/src/main/scala/wdl4s/wom/callable/TaskDefinition.scala:23]: This will go away with https://github.com/broadinstitute/wdl4s/pull/223

**Review by Horneth — COMMENTED** [review]
Taking a second look, it seems to me that the only WDL specific bit that is actually passed through is `findOutputFiles` which is really only calling `evaluateFiles` on all the output expressions. `evaluateFiles` is a method on `WomExpression` and already being provided by WDL through `WdlWomExpression` so I *think* we might not really need this ? :/

**Comment by danbills** [comment]
going to use approach suggested by @Horneth in https://github.com/broadinstitute/cromwell/pull/2656

**Pull request closed without merging**